src/renderer/components/Bonds/Bonds.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback } from 'react'
+
+import { Form } from 'antd'
+import { useIntl } from 'react-intl'
+
+import { Network } from '../../../shared/api/types'
+import { AddressValidation } from '../../services/thorchain/types'
+import { ValidatePasswordLD } from '../../services/wallet/types'
+import * as Styled from './Bonds.styles'
+import { Node } from './types'
+
+type Props = {
+  nodes: Node[]
+  removeNode: (node: string) => void
+  validatePassword$: (_: string) => ValidatePasswordLD
+  goToNode: (node: string) => void
+  network: Network
+  addNode: (node: string) => void
+  addressValidation: AddressValidation
+  className?: string
+}
+
+export const Bonds: React.FC<Props> = ({
+  addressValidation,
+  nodes,
+  removeNode,
+  validatePassword$,
+  goToNode,
+  network,
+  addNode,
+  className
+}) => {
+  const intl = useIntl()
+  const addressValidator = useCallback(
+    async (_: unknown, value: string) => {
+      if (!value) {
+        return Promise.reject(intl.formatMessage({ id: 'wallet.errors.address.empty' }))
+      }
+
+      const loweredCaseValue = value.toLowerCase()
+
+      if (!addressValidation(loweredCaseValue)) {
+        return Promise.reject(intl.formatMessage({ id: 'wallet.errors.address.invalid' }))
+      }
```

Comment:
yes, it will be used but in PR about business logic as this requires THORChain client which will be injected in `BondsView.tsx` and passed here as prop `addressValidation`

Reply:
Ok.